### PR TITLE
Add JWT authorization

### PR DESCRIPTION
**Pull request by apankratau**

### CODE REVIEW

File: module-2/middlewares/check-jwt.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Request, Response, NextFunction } from 'express';
+import AuthService from '../services/AuthService';
+
+export default function (req: Request, res: Response, next: NextFunction): void {
+  const auth = req.headers.authorization;
+
+  if (!auth) {
+    res.status(401).send({ message: 'Not authorized' });
+  } else {
```
Comment (Dmitry-White) [inline module-2/middlewares/check-jwt.ts]: "Else" part is redundant in term of a typical code style for Node.js controllers.
`if (!auth)` part looks like an "early return" pattern.

File: module-2/middlewares/check-jwt.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    const accessToken = auth
+      ?.split(' ')
+      .map(v => v.trim())
+      .filter(v => !/(bearer)/i.exec(v))[0];
+
+    if (AuthService.isAccessTokenValid(accessToken)) {
```
Comment (Dmitry-White) [inline module-2/middlewares/check-jwt.ts]: Consider making this part and all of the similar parts in auth router controllers similar to each other.
E.g. POST /login looks confusingly different from the rest of the controllers.

File: module-2/routers/auth.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      res.status(403).send({ message: 'Provided token is expired' });
+    }
+  }),
+);
+
+router.post('/logout', validators(AuthSchema.token, 'body'), (req, res) => {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const { token } = req.body;
+
+  AuthService.invalidateRefreshToken(token);
```
Comment (Dmitry-White) [inline module-2/routers/auth.ts]: I'm afraid this is not enough to successfully logout.
Basically, it's preventing user from getting access token without providing credentials.
However, previous access token might still be present on the client side until the expiration, thus enabling access to the services.
Consider either maintaining a blacklist of tokens or storing an encrypted token with the user in a DB.

**Review by Dmitry-White — CHANGES_REQUESTED** [review]

File: module-2/middlewares/check-jwt.ts (hunk 1 above)
Comment (apankratau) [reply module-2/middlewares/check-jwt.ts]: Fixed

File: module-2/middlewares/check-jwt.ts (hunk 2 above)
Comment (apankratau) [reply module-2/middlewares/check-jwt.ts]: Done, thanks!

File: module-2/routers/auth.ts (hunk 3 above)
Comment (apankratau) [reply module-2/routers/auth.ts]: Got it, switched to storing access tokens in DB

**Review by Dmitry-White — APPROVED** [review]

**Pull request merged**